### PR TITLE
Add postgresql15

### PR DESCRIPTION
**Pull request by lpsinger**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.1 21G217 x86_64
Xcode 14.0.1 14A400

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by dgilman** [comment]
I have been testing out a similar portfile locally, there are some changes that should be included:

- upstream dropped python 2.7 support, so that variant can be removed
- llvm 14 works, but 15 doesn't. the next point release of PG will likely have llvm 15 support, and it will be backported to previous supported versions. I added `-clang15` to `compilers.setup` to try and block llvm/clang 15, but that is sort of hacky and will probably go away quickly too.
- probably worth bumping the python3 variant to python 3.10, but I didn't personally test that

**Comment by lpsinger** [comment]
Should be all done now.

File: databases/postgresql15/Portfile [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+PortSystem 1.0
+PortGroup select 1.0
+PortGroup compilers 1.0
+PortGroup compiler_blacklist_versions 1.0
+PortGroup muniversal 1.0
+
+#remember to update the -doc and -server as well
+name                postgresql15
+version             15.0
+revision            1
```
Comment (dgilman) [inline databases/postgresql15/Portfile:12]: drop this line? i think it should be the implicit revision 0

**Comment by dgilman** [comment]
There is now a postgresql 15.1. I tested it out with LLVM 15 and it is working fine.

File: databases/postgresql15/Portfile (hunk 1 above)
Comment (lpsinger) [reply databases/postgresql15/Portfile]: Fixed.

**Comment by reneeotten** [comment]
I don't see a commitment here from @jyrkiwahlstedt to be the maintainer. 

So please set the ports to "nomaintainer" or volunteer to be the maintainer yourself and this can be merged.

**Comment by lpsinger** [comment]
> I don't see a commitment here from @jyrkiwahlstedt to be the maintainer.
> 
> So please set the ports to "nomaintainer" or volunteer to be the maintainer yourself and this can be merged.

Done.

**Pull request merged**